rust/gateway/src/eventloop.rs

```diff
@@ -24,6 +24,15 @@ use std::time::Duration;
 
 pub const PHOENIX_TOPIC: &str = "gateway";
 
+/// How long we allow a DNS resolution via `libc::get_addr_info`.
```

Comment:
So keep in mind that libc will default to timing out after 5s, and _then_ use a fallback server. So this might prevent the nameserver fallback?

Reply:
I think this might be ok though because the next request will presumably use the "next" nameserver right away (we can't keep trying nameservers forever).

Reply:
In that case, we should probably make it 10s?

Reply:
sgtm